frontend/public/monitoring.ts

```diff
@@ -60,3 +60,11 @@ const stateToProps = (desiredURLs: string[], state) => {
 };
 
 export const connectToURLs = (...urls) => connect(state => stateToProps(urls, state));
+
+// Determine if an Alert is silenced by a Silence (if all of the Silence's matchers match one of the Alert's labels)
+export const isSilenced = (alert, silence) => _.get(silence, 'status.state') === 'active' &&
+  _.every(silence.matchers, m => {
+    const alertValue = _.get(alert.labels, m.name);
+    return alertValue !== undefined &&
+      (m.isRegex ? (new RegExp(`^(?:${m.value})$`)).test(alertValue) : alertValue === m.value);
```

Comment:
It's not clear to me why we need the non-capturing group here since we're just testing for a match. Couldn't this just be

```suggestion
      (m.isRegex ? (new RegExp(`^${m.value}$`)).test(alertValue) : alertValue === m.value);
```